### PR TITLE
Settings Password And Data Wipe

### PR DESCRIPTION
**Pull request by jasonrogena**
Makes the settings screen password protected and adds a 'Wipe Data' button at the bottom of the settings screen

Fix for issue #35

### CODE REVIEW

File: src/main/java/com/onaio/steps/activities/SettingsActivity.java [hunk 1]
```diff
@@ -93,4 +89,15 @@ public void enableParticipantFlow(View view) {
         prepareViewWithData();
     }
 
+    public void eraseData(View view) {
+        DialogInterface.OnClickListener confirmListener = new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                db.truncate();
```
Comment (ukanga) [inline src/main/java/com/onaio/steps/activities/SettingsActivity.java]: Do we have a confirmation for the deletion? This sounds like where someone may need to confirm before the deletion is done. Or is it not required for this?

File: src/main/java/com/onaio/steps/helper/AuthDialog.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+     */
+    private void updateLastAuthTime() {
+        long now = Calendar.getInstance().getTimeInMillis();
+        KeyValueStoreFactory.instance(activity).putString(SETTINGS_AUTH_TIME, String.valueOf(now));
+    }
+
+    public boolean needsAuth() {
+        long lastAuthTime = getLastAuthTime();
+        long now = Calendar.getInstance().getTimeInMillis();
+        if(lastAuthTime <= now && (now - 30000) < lastAuthTime) {
```
Comment (ukanga) [inline src/main/java/com/onaio/steps/helper/AuthDialog.java]: What is this `30000`? Should it be a static or configurable field?

File: src/main/java/com/onaio/steps/activities/SettingsActivity.java (hunk 1 above)
Comment (jasonrogena) [inline src/main/java/com/onaio/steps/activities/SettingsActivity.java]: The DialogInterface presents to the user the confirmation dialog

File: src/main/java/com/onaio/steps/helper/AuthDialog.java (hunk 2 above)
Comment (jasonrogena) [inline src/main/java/com/onaio/steps/helper/AuthDialog.java]: True. That's the number of milliseconds before the authorization expires in the settings screen

File: src/main/java/com/onaio/steps/activities/SettingsActivity.java (hunk 1 above)
Comment (ukanga) [inline src/main/java/com/onaio/steps/activities/SettingsActivity.java]: Cool.

**Pull request merged**